src/System.Runtime/tests/System/String.Split.cs

```diff
@@ -0,0 +1,515 @@
+
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitEmptyValueWithRemoveEmptyEntriesOptionEmptyResult()
+    {
+        string value = string.Empty;
+        const int count = int.MaxValue;
+        const StringSplitOptions options = StringSplitOptions.RemoveEmptyEntries;
+
+        string[] expected = new string[0];
+
+        Assert.Equal(expected, value.Split(',', options));
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(",", options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitOneCountSingleResult()
+    {
+        const string value = "a,b";
+        const int count = 1;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        string[] expected = new[] { value };
+
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitNoMatchSingleResult()
+    {
+        const string value = "a b";
+        const int count = int.MaxValue;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        string[] expected = new[] { value };
+
+        Assert.Equal(expected, value.Split(','));
+        Assert.Equal(expected, value.Split(',', options));
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }));
+        Assert.Equal(expected, value.Split(new[] { ',' }, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(","));
+        Assert.Equal(expected, value.Split(",", options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    private const int M = int.MaxValue;
+
+    [Theory]
```

Comment:
Yes this is my attempt to get @ellismg to fix our internal system as opposed to stopping folks from using it.